### ISSUE TITLE
Numeric encoding of dependent variables stored as string

### ISSUE THREAD

**Issue by loicdtx** [CONABIO/antares3#10]
Training data is in many cases stored as string (e.g.: `forest`, `agriculture`, `water`, ...).  Pixel based prediction requires that categories be numerical. We therefore need a mechanism to encode dependent variables. RIght now this is done in a hackish way using sklearn `LabelEncoder`. A better way would be to have a dictionary (stored as json in the database) of label:code key pair values for each `TrainTag` of each `TrainingSet`.

Also encoding may be applied after prediction (https://github.com/CONABIO/antares3/blob/2e9cac04fab1b80f950a309344f599ac72dc72cf/madmex/wrappers.py#L76), instead of before model fitting, to allow a single model to predict both pixel based and object based.

**Comment by loicdtx** [comment]
Starting at 5dd19e77723a450c2fbc25e6759273bf6beeeb0c all modeling is done using a numeric dependent variable. The issue is therefore reverse now: decode numeric for text based prediction when running object based prediction

**Comment by loicdtx** [comment]
`madmex_tag.id` is now used as dependent variable for modeling. No encoding/decoding is required for object based classification, but a it would be preferable to encode pixel based predictions
